plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestJdbcPageSink.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.jdbc;
+
+import io.trino.RowPagesBuilder;
+import io.trino.spi.Page;
+import io.trino.spi.type.BigintType;
+import io.trino.spi.type.Type;
+import io.trino.spi.type.VarcharType;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.ThreadLocalRandom;
+
+import static io.trino.RowPagesBuilder.rowPagesBuilder;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@Test(singleThreaded = true)
+public class TestJdbcPageSink
+{
+    private static final String TABLE_SQL = "CREATE TABLE example.dummy (txt varchar(36), num bigint)";
+    private static final String INSERT_SQL = "INSERT INTO example.dummy(txt, num) VALUES (?, ?)";
+    private static final List<Type> COLUMN_TYPES = Arrays.asList(VarcharType.VARCHAR, BigintType.BIGINT);
+    private static final List<WriteFunction> COLUMN_WRITERS = Arrays.asList(
+            (SliceWriteFunction) (statement, index, value) -> statement.setString(index, value.toStringUtf8()),
+            (LongWriteFunction) PreparedStatement::setLong);
+
+    private TestingDatabase db;
+    private Connection connection;
+    private PreparedStatement statement;
+
+    @BeforeMethod
+    public void setUp() throws SQLException
+    {
+        db = new TestingDatabase();
+        connection = db.getConnection();
+        connection.createStatement().execute(TABLE_SQL);
+        statement = spy(connection.prepareStatement(INSERT_SQL));
```

Comment:
We are not using either spy or mocks (thus no mockito in the codebase).

Reply:
Just make an e2e test that ensures that given session with a "write batch size" property set to a given `n`, after inserting `m=100` rows still, make them readable after insert (verify with `count`).

Reply:
For referring to existing e2e tests take a look at BaseJdbcConnectorTest.

Reply:
Removing mockito, suggesting to keep it as a unit-test with explicit select and verify.